frontend/src/static/js/services/webstatus-firebase-auth-service.ts

```diff
@@ -84,8 +117,45 @@ export class WebstatusFirebaseAuthService extends ServiceElement {
       // Set up the callback that will detect when:
       // 1. The user first logs in
       // 2. Resuming a session
-      this.firebaseAuthConfig.auth.onAuthStateChanged(user => {
-        this.user = user ? user : undefined;
+      this.firebaseAuthConfig.auth.onAuthStateChanged((user: User | null) => {
+        if (user !== null) {
+          this.user =
+            user !== null ? {...user, gitHubUsername: 'empty'} : undefined;
+          this._loadingGithubUsername = new Task(this, {
```

Comment:
We will continue to make a request for the information. However, we should:
1. use [octokit](https://github.com/octokit/octokit.js). This should simplify the retry logic since the client handles it automatically.
2. Try to do it without the Task. We actually shouldn't need it. That should make this a lot simpler too.

For step 1: run `cd frontend && npm install octokit`